tests/lib/rules/no-child-content.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * @author Flo Edelmann
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const RuleTester = require('eslint').RuleTester
+const rule = require('../../../lib/rules/no-child-content')
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+const ruleTester = new RuleTester({
+  parser: require.resolve('vue-eslint-parser'),
+  parserOptions: { ecmaVersion: 2015 }
+})
+
+ruleTester.run('no-child-content', rule, {
+  valid: [
+    {
+      // element without directive
+      filename: 'test.vue',
+      code: '<template><div>foo</div></template>'
+    },
+    {
+      // element with unknown directive
+      filename: 'test.vue',
+      code: '<template><div v-foo="bar" /></template>'
+    },
+    {
+      // self-closing element with v-html directive
+      filename: 'test.vue',
+      code: '<template><div v-html="foo" /></template>'
+    },
+    {
+      // self-closing element with v-text directive
+      filename: 'test.vue',
+      code: '<template><div v-text="foo" /></template>'
+    },
+    {
+      // empty element with v-html directive
+      filename: 'test.vue',
+      code: '<template><div v-html="foo"></div></template>'
+    },
+    {
+      // self-closing element with v-t directive
+      filename: 'test.vue',
+      options: [{ additionalDirectives: ['t'] }],
+      code: '<template><div v-t="foo" /></template>'
+    }
+  ],
+  invalid: [
+    {
+      // v-html directive and text content
+      filename: 'test.vue',
+      code: '<template><div v-html="foo">bar</div></template>',
+      errors: [
+        {
+          message:
+            'Child content is disallowed because it will be overwritten by the v-html directive.',
+          column: 29,
+          endColumn: 32,
+          suggestions: [
+            { output: '<template><div v-html="foo"></div></template>' }
+          ]
+        }
+      ]
+    },
+    {
+      // v-html directive and whitespace-only text content
+      filename: 'test.vue',
+      code: '<template><div v-html="foo"> </div></template>',
```

Comment:
I just noticed this test case 😓.
I think it's better not to report in the case of whitespace or indentation. It's consistent with the vue compiler.


https://sfc.vuejs.org/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgZm9vID0gcmVmKCdIZWxsbyBXb3JsZCEnKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT48ZGl2IHYtaHRtbD1cImZvb1wiPiA8L2Rpdj48L3RlbXBsYXRlPiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vc2ZjLnZ1ZWpzLm9yZy92dWUucnVudGltZS5lc20tYnJvd3Nlci5qc1wiXG4gIH1cbn0ifQ==

Reply:
That's a helpful website. Let's allow whitespace contents then, and report comments, to stay consistent with the Vue compiler. I'll add the relevant test cases.